tests/test_schemas.py

```diff
@@ -274,8 +290,169 @@ def test_generate_node(self):
+        before_edges = deepcopy(G.edges)
+
+        # Generate edges for whichever node we are testing
+        G = DMEdges.generate_edge(G, node_to_add, all_node_dict, parsed_data_model, edge_relationships)
+
+        # Assert that the current edges are different from the edges of the graph before
+        assert G.edges > before_edges
+
+        # Assert that somewhere in the current edges for the node we added, that the correct relationship exists
+        relationship_df = pd.DataFrame(G.edges, columns= ['node1', 'node2', 'edge'])
+        assert (relationship_df['edge'] == edge_relationship).any()
+        
+        return
+    
+    @pytest.mark.parametrize("node_to_add, other_node, expected_weight, data_model_path", 
+                             [("Patient ID", "Biospecimen", 1, "validator_dag_test.model.csv"),
+                              ("dataset_id", "cohorts", -1, "properties.test.model.csv")],
+                              ids=["list", "domainIncludes"])
+    def test_generate_weights(self, helpers, DMR, DMEdges, node_to_add, other_node, expected_weight, data_model_path):
+        # Instantiate graph object
+        G = nx.MultiDiGraph()
+
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_path)
+
+        #Parse Model
+        parsed_data_model = data_model_parser.parse_model()
+
+        # Instantiate data model Nodes object
+        dmn = DataModelNodes(parsed_data_model)
+
+        # Get edge relationships and all nodes from the parsed model
+        edge_relationships = DMR.define_edge_relationships()
+        all_nodes = dmn.gather_all_nodes(attr_rel_dict=parsed_data_model)
+
+        # Sanity check to ensure that the node we intend to test exists in the data model
+        assert node_to_add in all_nodes
+        
+        # Add all nodes to the graph 
+        all_node_dict = {}
+        for node in all_nodes:
+            node_dict = dmn.generate_node_dict(node, parsed_data_model)
+            all_node_dict[node] = node_dict
+            G = dmn.generate_node(G, node_dict)
+
+        # Check the edges in the graph, there should be none
+        before_edges = deepcopy(G.edges)
+
+        # Generate edges for whichever node we are testing
+        G = DMEdges.generate_edge(G, node_to_add, all_node_dict, parsed_data_model, edge_relationships)
+
+        # Assert that the current edges are different from the edges of the graph before
+        assert G.edges > before_edges
+        
+        # Cast the edges and weights to a DataFrame for easier indexing
+        edges_and_weights = pd.DataFrame(G.edges.data(), columns= ['node1', 'node2', 'weights']).set_index('node1')
+
+        # Get the index of the property in the schema
+        # Weights for properties are determined by their order in the schema. 
+        # This would allow the tests to continue to function correctly in the case were other attributes were added to the schema
+        if expected_weight < 0:
```

Comment:
Here can you indicate that you specifically set the expected weight to -1 to offer this chance for a lookup?

Reply:
yep!